### PR TITLE
Features:add Profile Settings

### PR DESCRIPTION
**Pull request by igmeMarcial**
image
image
Check my code to see if it is as optimized as possible or if the code flow is not optimized. There are small things to modify, in addition to the path, which is currently in /settings.

### CODE REVIEW

**Comment by igmeMarcial** [comment]
I haven't tried with next build yet, it doesn't create anything in api/

File: src/app/(public)/settings/page.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import { getUser } from '@/lib/auth/lucia'
+import { getUserProfile } from '@/services/actions/userActions'
+import { redirect } from 'next/navigation'
+import React from 'react'
+async function page() {
+  const sesion = await getUser()
+  if (!sesion?.id) {
+    redirect('/auth')
+  }
+  const user = await getUserProfile()
```
Comment (pbrain19) [inline src/app/(public)/settings/page.tsx:11]: no need for 

```
const sesion = await getUser()
  if (!sesion?.id) {
    redirect('/auth')
  }
```

this is because getUserProfile already checks for this

File: src/app/api/profile/route.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { NextRequest, NextResponse } from 'next/server'
+
+import { updateUserProfile } from '@/services/actions/userActions'
```
Comment (pbrain19) [inline src/app/api/profile/route.ts:3]: No need for this file. We use server actions instead.

check the services/actions folder and see how we use this.

The main reason is that is strongly typed.

File: src/components/profile-settings/delete-account-dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
+'use client'
+import { useState } from 'react'
```
Comment (pbrain19) [inline src/components/profile-settings/delete-account-dialog.tsx:2]: Wrong naming convention for the file. It should be DeleteAccountDialog.tsx

File: src/components/profile-settings/delete-account-dialog.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+import { useRouter } from 'next/navigation'
+import { deleteUserAccount } from '@/services/actions/userActions'
+
+export default function DeleteAccountDialog() {
+  const [isOpen, setIsOpen] = useState(false)
+  const [isLoading, setIsLoading] = useState(false)
+  const { toast } = useToast()
+  const router = useRouter()
+
+  const handleDelete = async () => {
```
Comment (pbrain19) [inline src/components/profile-settings/delete-account-dialog.tsx:28]: Wrap this in a usecallback

File: src/components/profile-settings/form-field-wrapper.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  form: UseFormReturn<ProfileFormValues>
+  name: keyof ProfileFormValues
+  label: string
+  placeholder: string
+  icon: LucideIcon
+  disabled?: boolean
+  type?: string
+}
+
+export default function FormFieldWrapper({
```
Comment (pbrain19) [inline src/components/profile-settings/form-field-wrapper.tsx:24]: I am not entirely sure this file is required. Check the event creation form for inspiration.

File: src/components/profile-settings/profile-form.tsx [hunk 6]
```diff
@@ -0,0 +1,189 @@
+import { UseFormReturn } from 'react-hook-form'
```
Comment (pbrain19) [inline src/components/profile-settings/profile-form.tsx:1]: Please look at how we use zod in our forms

src/components/events/createEventForm.tsx

File: src/components/profile-settings/profile-settings.tsx [hunk 7]
```diff
@@ -0,0 +1,139 @@
+'use client'
+import { useMemo, useRef, useState } from 'react'
```
Comment (pbrain19) [inline src/components/profile-settings/profile-settings.tsx:2]: rename file and view the example form src/components/events/createEventForm.tsx

File: src/constantes/form.ts [hunk 8]
```diff
@@ -0,0 +1,5 @@
+export const FORM_SECTIONS = {
```
Comment (pbrain19) [inline src/constantes/form.ts:1]: why is this required?

i am not sure we need this at all.

File: src/services/actions/userActions.ts [hunk 9]
```diff
@@ -32,3 +35,108 @@ export async function makeRegularUser(user: { userId: string }) {
     revalidatePath('/', 'layout')
   }
 }
+
+export async function updateUserProfile(
+  data: ProfileDataValues,
+) {
+   try {
```
Comment (pbrain19) [inline src/services/actions/userActions.ts:42]: 2 things are missing.

1. u should always validate the input data by using zod.parse. Checkout other actions.
2. u must validate that the current user is the owner of that profile.

File: src/types/user.ts [hunk 10]
```diff
@@ -0,0 +1,19 @@
+
+
+export interface UserType {
```
Comment (pbrain19) [inline src/types/user.ts:3]: We should not be using types like this... use zod. checkout the schemas folder.

**Review by pbrain19 — CHANGES_REQUESTED** [review]
Many changes are required but good first draft.

**Comment by pbrain19** [comment]
also add a country code for phone number
